wallet/src/components/IbcTransfer.tsx

```diff
@@ -0,0 +1,491 @@
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogTitle from '@mui/material/DialogTitle';
+import { parseAsValue } from '@agoric/ui-components';
+import { withApplicationContext } from '../contexts/Application';
+import Petname from './Petname';
+import { ibcAssets } from '../util/ibc-assets';
+import { stringifyPurseValue } from '@agoric/ui-components';
+import ArrowDownward from '@mui/icons-material/ArrowDownward';
+import { Box } from '@mui/system';
+import { fromBech32 } from '@cosmjs/encoding';
+import { queryBankBalances } from '../util/queryBankBalances';
+import type { PurseInfo } from '../service/Offers';
+import type { KeplrUtils } from '../contexts/Provider';
+import { SigningStargateClient } from '@cosmjs/stargate';
+import { CircularProgress, Link, Snackbar, Typography } from '@mui/material';
+
+export enum IbcDirection {
+  Deposit,
+  Withdrawal,
+}
+
+const titlePreposition = {
+  [IbcDirection.Deposit]: 'from',
+  [IbcDirection.Withdrawal]: 'to',
+};
+
+const agoricAddressLabel = {
+  [IbcDirection.Deposit]: 'To',
+  [IbcDirection.Withdrawal]: 'From',
+};
+
+const remoteChainAddressLabel = {
+  [IbcDirection.Deposit]: 'From',
+  [IbcDirection.Withdrawal]: 'To',
+};
+
+interface Params {
+  purse?: PurseInfo;
+  direction: IbcDirection;
+  handleClose: () => void;
+  keplrConnection: KeplrUtils;
+}
+
+const secondsUntilTimeout = 300;
+
+const timeoutTimestampSeconds = () =>
+  Math.round(Date.now() / 1000) + secondsUntilTimeout;
+
+const agoricChainId = 'agoric-3';
+const agoricRpc = 'https://agoric-rpc.stakely.io/';
+
+// Exported for testing only.
+export const IbcTransferInternal = ({
+  purse,
+  handleClose,
+  direction,
+  keplrConnection,
+}: Params) => {
+  const ibcAsset =
+    typeof purse?.brandPetname === 'string'
+      ? ibcAssets[purse.brandPetname]
+      : undefined;
+
+  const purseBalance = purse?.currentAmount.value;
+
+  const [inProgress, setInProgress] = useState(false);
+  const [error, setError] = useState('');
+  const [amount, setAmount] = useState('');
+  const [remoteChainAddress, setRemoteChainAddress] = useState('');
+  const [remoteChainBalance, setRemoteChainBalance] = useState<bigint | null>(
+    null,
+  );
+  const [remoteChainSigner, setRemoteChainSigner] = useState(null);
+
+  const [isSnackbarOpen, setIsSnackbarOpen] = useState(false);
+  const handleCloseSnackbar = _ => {
+    setIsSnackbarOpen(false);
+  };
+  const [snackbarMessage, setSnackbarMessage] = useState('');
+  const showSnackbar = msg => {
+    setSnackbarMessage(msg);
+    setIsSnackbarOpen(true);
+  };
+
+  const isRemoteChainAddressInvalid = useMemo(() => {
+    if (!remoteChainAddress) return true;
+
+    try {
+      const { prefix } = fromBech32(remoteChainAddress);
+      return prefix !== ibcAsset?.chainInfo.addressPrefix;
+    } catch (e) {
```

Comment:
lint rules allow unused vars without a special name like `_e`?

Reply:
Apparently, just fixed this instance though